plasoscaffolder/frontend/frontend.py

```diff
@@ -0,0 +1,373 @@
+# !/usr/bin/python
+# -*- coding: utf-8 -*-
+"""The scaffolder frontend."""
+import re
+
+from typing import Dict
+from typing import List
+from typing import Type
+
+from plasoscaffolder.frontend import output_handler
+from plasoscaffolder.helpers import git
+from plasoscaffolder.lib import engine
+from plasoscaffolder.lib import errors
+
+from plasoscaffolder.definitions import interface as definition_interface
+from plasoscaffolder.definitions import manager as definition_manager
+from plasoscaffolder.scaffolders import interface as scaffolder_interface
+from plasoscaffolder.scaffolders import manager as scaffolder_manager
+
+
+class ScaffolderFrontend:
+  """A frontend implementation for the scaffolder project.
+
+  Attributes:
+    OUTPUT_HANDLER (output_handler.BaseOutputHandler: output handler
+        that is used to request and read input from end user.
+  """
+
+  OUTPUT_HANDLER = output_handler.BaseOutputHandler()
+
+  @classmethod
+  def _AskDictQuestion(
+      cls, question: scaffolder_interface.BaseQuestion) -> Dict[str, Type]:
+    """Ask the user a question and return a dict answer back.
+
+    Args:
+      question (scaffolder_interface.BaseQuestion): the question to ask.
+
+    Returns:
+      dict: the answer as supplied by the user.
+    """
+    return_dict = {}
+
+    more_entries = True
+    entry_index = 1
+    while more_entries:
+      # pylint: disable=assignment-from-no-return
+      key = cls.OUTPUT_HANDLER.PromptInfo(
+          '{0:s} [#{1:d}]'.format(question.key_prompt, entry_index))
+      if key in return_dict:
+        cls.OUTPUT_HANDLER.PrintError(
+            'Key value ({0:s}) already entered.'.format(key))
+        continue
+
+      value = cls.OUTPUT_HANDLER.PromptInfo(
+          '{0:s} [#{1:d}]'.format(question.value_prompt, entry_index))
+
+      return_dict[key] = value
+      more_entries = cls.OUTPUT_HANDLER.Confirm('More entries?', abort=False)
+      entry_index += 1
+
+    question.ValidateAnswer(return_dict)
+    return return_dict
+
+  @classmethod
+  def _AskListQuestion(
+      cls, question: scaffolder_interface.BaseQuestion) -> List[object]:
```

Comment:
Shouldn't this be ListQuestion, not BaseQuestion?

Reply:
And since PromptInfo returns a str, shouldn't this return a list of strings, not objects?